src/map.js

```diff
@@ -66,99 +75,148 @@ export default class GpxMap {
                 stateName: 'default',
                 title: 'Open settings dialog',
                 onClick: (_btn, map) => {
-                    buildSettingsModal(this.options, (opts) => {
+                    buildSettingsModal(this, this.options, (opts) => {
```

Comment:
Seems like only `this.tracks` is needed, could you pass that to `buildSettingsModal` instead of the entire `this`?

Reply:
Fixed in 51a4573eebb917719c0a68222a03fe72ed6a2d8f